flask_dance/contrib/twitch.py

```diff
@@ -0,0 +1,111 @@
+    authorization_url="https://id.twitch.tv/oauth2/authorize",
+    token_url="https://id.twitch.tv/oauth2/token",
+    token_url_params={"include_client_id": True},
+    authorized_url=None,
+    session_class=None,
+    storage=None,
+):
+    """Make a blueprint for authenticating with Twitch using OAuth 2.
+
+    This requires a client ID and client secret from Twitch.
+    You should either pass them to this constructor, or make sure that your Flask application config defines them, using the variables :envvar:`TWITCH_OAUTH_CLIENT_ID` and :envvar:`TWITCH_OAUTH_CLIENT_SECRET`.
+
+    Args:
+        client_id (str): The client ID for your application on Twitch. Defaults to app config "TWITCH_OAUTH_CLIENT_ID".
+        client_secret (str): The client Secret for your application on Twitch. Defaults to app config "TWITCH_OAUTH_CLIENT_SECRET".
+        scope (list, optional): Comma-separated list of scopes for the OAuth token.Defaults to None.
+        redirect_url (str, optional): The URL to redirect to after the authentication dance is complete. Defaults to None.
+        redirect_to (str, optional): If ``redirect_url`` is not defined, the name of the view to redirect to after the authentication dance is complete. Defaults to None.
+        login_url (str, optional): The URL path for the ``login`` view. Defaults to None.
+        base_url (str, optional): Base URL for the Twitch API. Defaults to "{Twitch API}/helix/".
+        authorization_url (str, optional): URL for the Twitch API login endpoint. Defaults to "{Twitch API}/oauth2/authorize".
+        token_url (str, optional): URL for Twitch token API. Defaults to "{Twitch API}/oauth2/token".
+        token_url_params (dict, optional): Options to send to the token URL. Defaults to {"include_client_id": True}.
+        authorized_url (str, optional): The URL path for the ``authorized`` view. Defaults to None.
+        session_class (class, optional):  The class to use for creating a Requests session. Defaults to None.
+        storage (class, optional): A token storage class, or an instance of a token storage class, to use for this blueprint. Defaults to None.
+
+    Raises:
+        MissingConfigException: If client_id or client_secret is missing when request is made.
+
+    Returns:
+        :class:`~flask_dance.consumer.OAuth2ConsumerBlueprint` A :ref:`blueprint <flask:blueprints>` to attach to your Flask app.
+    """
+    twitch_bp = OAuth2ConsumerBlueprint(
+        "twitch",
+        __name__,
+        client_id=client_id,
+        client_secret=client_secret,
+        scope=scope,
+        base_url=base_url,
+        authorization_url=authorization_url,
+        token_url=token_url,
+        token_url_params=token_url_params,
+        redirect_url=redirect_url,
+        redirect_to=redirect_to,
+        login_url=login_url,
+        authorized_url=authorized_url,
+        session_class=session_class,
+        storage=storage,
+    )
+    if twitch_bp.client_id is None:
```

Comment:
You don't need these `if` statements here. The `from_config` values will only be used if the values are set in the app config; otherwise, they will be ignored. See https://github.com/singingwolfboy/flask-dance/blob/main/flask_dance/consumer/base.py#L80

Reply:
Perfect!